requirements-dev.txt

```diff
@@ -51,11 +53,11 @@ dask
 seaborn
 moto
 flask
-asv>=0.5.1
-flake8==6.0.0
-mypy==1.4.1
+asv>=0.6.1
+flake8==6.1.0
+mypy==1.7.1
```

Comment:
This mypy change requires more changes :)

I have a branch to update mypy and pyright https://github.com/twoertwein/pandas/tree/typing_deps. I'm waiting for the next pyright release to fix https://github.com/microsoft/pyright/issues/6695 and will then open a PR.